pkg/functions/secretinjector/secretinjector.go

```diff
@@ -8,23 +8,31 @@ package secretinjector
 import (
 	"context"
 
+	"github.com/vmware/dispatch/pkg/entity-store"
+
 	apiclient "github.com/go-openapi/runtime/client"
 
 	"github.com/pkg/errors"
 	"github.com/vmware/dispatch/pkg/functions"
 	secretclient "github.com/vmware/dispatch/pkg/secret-store/gen/client"
 	"github.com/vmware/dispatch/pkg/secret-store/gen/client/secret"
+	serviceclient "github.com/vmware/dispatch/pkg/service-manager/gen/client"
+	serviceinstance "github.com/vmware/dispatch/pkg/service-manager/gen/client/service_instance"
 
 	log "github.com/sirupsen/logrus"
 )
 
 type secretInjector struct {
-	client *secretclient.SecretStore
+	secretClient  *secretclient.SecretStore
+	serviceClient *serviceclient.ServiceManager
 }
 
 // New create a new secret injector
-func New(secretClient *secretclient.SecretStore) functions.SecretInjector {
-	return &secretInjector{client: secretClient}
+func New(secretClient *secretclient.SecretStore, serviceClient *serviceclient.ServiceManager) functions.SecretInjector {
```

Comment:
Could we define a new injector for service bindings here? This way secretinjector can remain as is, and we will have better separation of concerns.

Reply:
I was going to go that route... but I need to call the secrets API from the services injector also.  I can still make this change if you like.